internal/datasources/web.go

```diff
@@ -0,0 +1,57 @@
+package datasources
+
+import (
+	"bytes"
+	"io"
+	"mime"
+	"net/http"
+)
+
+type WebFileDatasource struct {
+	uri string
+}
+
+func NewWebFileDatasource(uri string) *WebFileDatasource {
+	return &WebFileDatasource{uri}
+}
+
+func (ds *WebFileDatasource) Load() (map[string]any, error) {
+	data := make(map[string]any)
+
+	res, err := http.Get(ds.uri)
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+
+	ct := res.Header.Get("Content-Type")
+	mt, _, _ := mime.ParseMediaType(ct)
+
+	b, err := io.ReadAll(res.Body)
```

Comment:
no need to read the body and then create a buffer from it, as `res.Body` is already of type `io.ReadCloser` so it implements `io.Reader`.
you can directly pass `res.Body` to the datasource constructors